src/Analyser/MutatingScope.php

```diff
@@ -248,6 +248,13 @@ public function enterDeclareStrictTypes(): self
 			null,
 			null,
 			$this->expressionTypes,
+			[],
+			null,
+			null,
+			true,
+			[],
+			[],
+			$this->nativeExpressionTypes,
```

Comment:
maybe this one is not correct too. There should be no situation that native type assign before declare